### PR TITLE
Fix #752: Return activation lists with limits and paging

### LINKED ISSUE #752: Return activation lists with limits and paging
Currently, we return one huge page for per-user activations, we should limit the size and introduce paging.

We should use the standard Spring paging using Pageable: https://docs.spring.io/spring-data/rest/docs/current/reference/html/paging-chapter.html

See: https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/domain/Pageable.html

Paging should have two options:
- `pageNumber` - zero-based number of the page
- `pageSize` - size of each page

This change has impact into:
- REST API - add the parameters into the request as optional

Default values:
- `pageNumber`: `0`
- `pageSize`: `100`

Document the changes and implement basic tests. End-to-end tests of the API will be updated in https://github.com/wultra/powerauth-tests.

### PR DESCRIPTION
**Pull request by jandusil**
- Add Pageable properties as request params of getActivationList endpoint
- Propagate Pageable config to repository queries

### CODE REVIEW

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/PowerAuthController.java [hunk 1]
```diff
@@ -160,18 +159,28 @@ public ObjectResponse<RemoveActivationResponse> removeActivation(@RequestBody Ob
...
+     * @param pageSize This is an optional parameter that specifies the size of each page for the paginated results.
+     *                 It defaults to 100, which means each page will show up to 100 activations.
+     * @return This endpoint returns an {@link ObjectResponse} that contains a {@link GetActivationListForUserResponse},
+     *         which includes the list of activations for the given user and application ID.
+     * @throws Exception In case the service throws an exception, it will be propagated and should be handled by the caller.
      */
     @PostMapping("/activation/list")
-    public ObjectResponse<GetActivationListForUserResponse> getActivationListForUser(@RequestBody ObjectRequest<GetActivationListForUserRequest> request) throws Exception {
-        return new ObjectResponse<>("OK", powerAuthService.getActivationListForUser(request.getRequestObject()));
+    public ObjectResponse<GetActivationListForUserResponse> getActivationListForUser(@RequestBody ObjectRequest<GetActivationListForUserRequest> request,
```
[the remark this answers is not in the document]
Comment (banterCZ) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/PowerAuthController.java:177]: 🤔 But when I look into `com.wultra.app.enrollmentserver.api.model.enrollment.request.GetInboxListRequest`, we use `page` and `size` as a part of the request object.

**Comment by jandusil** [comment]
Connected to [https://github.com/wultra/powerauth-server/issues/752](url)

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/configuration/PowerAuthPageableConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2022 Wultra s.r.o.
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/configuration/PowerAuthPageableConfiguration.java:3]:
```suggestion
 * Copyright (C) 2023 Wultra s.r.o.
```

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/configuration/PowerAuthPageableConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * PowerAuthPageableConfiguration is a configuration class for handling pagination settings.
+ *
+ * @author Jan Dusil, jan.dusil@wultra.com
+ *
+ */
+@Configuration
+@ConfigurationProperties("powerauth.service.pagination")
+@Data
+public class PowerAuthPageableConfiguration {
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/configuration/PowerAuthPageableConfiguration.java:33]: 🤔 Just thinking whether we may start using `@ConstructorBinding` and _records_, when we already upgraded to JDK 17 https://www.baeldung.com/configuration-properties-in-spring-boot#java-16records

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/PowerAuthService.java [hunk 4]
```diff
@@ -146,8 +156,11 @@ public GetActivationListForUserResponse getActivationListForUser(GetActivationLi
         try {
             final String userId = request.getUserId();
             final String applicationId = request.getApplicationId();
+            final int pageNumber = request.getPageNumber() != null ? request.getPageNumber() : powerAuthPageableConfiguration.getDefaultPageNumber();
+            final int pageSize = request.getPageSize() != null ? request.getPageSize() : powerAuthPageableConfiguration.getDefaultPageSize();
+            Pageable pageable = PageRequest.of(pageNumber, pageSize);
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/PowerAuthService.java:161]:
```suggestion
            final Pageable pageable = PageRequest.of(pageNumber, pageSize);
```

File: powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/model/request/GetActivationListForUserRequest.java [hunk 5]
```diff
@@ -30,5 +30,7 @@ public class GetActivationListForUserRequest {
 
     private String userId;
     private String applicationId;
+    private Integer pageNumber;
```
Comment (banterCZ) [inline powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/model/request/GetActivationListForUserRequest.java:33]: May we start using bean validation?

File: powerauth-java-server/src/main/resources/application.properties [hunk 6]
```diff
@@ -105,6 +105,10 @@ spring.application.name=powerauth-java-server
 spring.cloud.vault.enabled=false
 spring.cloud.vault.kv.enabled=true
 
+# Configure Spring Pageable properties
+powerauth.service.pagination.default-page-size=100
```
Comment (banterCZ) [inline powerauth-java-server/src/main/resources/application.properties:109]: Most probably should be mentioned in `Configuration-Properties.md`

**Review by banterCZ — COMMENTED** [review]
Good job, I have some ideas on how to improve it a little bit.

**Review by banterCZ — APPROVED** [review]
Looks good.

**Pull request merged**